clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -1546,6 +1547,7 @@ private AppendCallbacks(Callback userCallback, ProducerInterceptors<K, V> interc
             // whole lifetime of the batch.
             // We don't want to have an NPE here, because the interceptors would not be notified (see .doSend).
             topic = record != null ? record.topic() : null;
+            headers = record != null ? record.headers() : new RecordHeaders();
```

Comment:
To maintain header immutability, especially given your efforts in this area, it might be prudent to pass immutable headers when record is null.

Reply:
miss this comment.

good point, I will update it

Reply:
fca85102d87580a593098b8c1eb5ee97d941a43f